modules/library/coverage/src/main/java/org/geotools/image/ImageWorker.java

```diff
@@ -644,6 +645,12 @@ public boolean isNoDataNeeded() {
      */
     public final ImageWorker setROI(final ROI roi) {
         this.roi = roi;
+        // If ROI == null remove it also from the image properties
+        if(roi == null){
```

Comment:
Formatting... not going to repeat this observation over and over, please check the rest of the files yourself thank you :)